src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -371,19 +370,27 @@ export class DataResourceDataProvider implements IGridDataProvider {
 	}
 
 	private transformSource(source: IDataResource): void {
-		this._rows = source.data.map(row => {
-			let rowData: azdata.DbCellValue[] = [];
-			Object.keys(row).forEach((val, index) => {
-				let displayValue = String(values(row)[index]);
-				// Since the columns[0] represents the row number, start at 1
-				rowData.push({
-					displayValue: displayValue,
-					isNull: false,
-					invariantCultureDisplayValue: displayValue
-				});
+		if (source.data.length > 0) {
+			let columns = source.schema.fields;
+			// Check if column names match the keys in the data. If they don't, then fall back to using ordinal keys for the data.
+			let useColumnNameKey = equals(columns.map(column => column.name), Object.keys(source.data[0]));
```

Comment:
Are these always guaranteed to be in the same order?

Also - couldn't we just check the type instead? If it's using the column names it'll be a string - where as if it's the index it'll be a number.

Reply:
Not sure if checking for the type will be sufficient here. Array indexing works the same with array[0] and array['0']. In the debugger it shows up as a number, but the ipynb file serializes the keys as strings. Checking for the presence of one of the column names might be sufficient, since it's either a number or a name in the keys.